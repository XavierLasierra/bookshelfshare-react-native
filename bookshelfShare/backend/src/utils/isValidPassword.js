const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
