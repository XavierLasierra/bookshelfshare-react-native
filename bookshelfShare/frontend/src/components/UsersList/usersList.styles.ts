import { StyleSheet } from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  marginBottom: {
    width: '100%',
    height: 130
  },
  usersListContainer: {
    paddingHorizontal: stylesConstants.padding.normal
  },
  clearSearchButton: {
    alignItems: 'center',
    justifyContent: 'center',
    borderTopWidth: 2,
    borderBottomWidth: 2,
    borderColor: stylesConstants.colors.secondary,
    height: 40,
    width: 100,
    alignSelf: 'center',
    marginBottom: 10
  },
  clearSearchButtonText: {
    fontSize: stylesConstants.fontSize.mid
  },
  resultsContainer: {
    alignItems: 'center'
  }
});
