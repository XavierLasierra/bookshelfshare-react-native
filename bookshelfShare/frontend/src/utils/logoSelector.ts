import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
