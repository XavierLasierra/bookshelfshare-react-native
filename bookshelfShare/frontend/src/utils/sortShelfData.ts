export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
