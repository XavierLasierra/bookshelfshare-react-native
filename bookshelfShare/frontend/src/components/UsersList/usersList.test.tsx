import React from 'react';
import UsersList from './UsersList';
import {render, fireEvent} from '../../utils/test.utils';
import loggedUserMock from '../../mocks/loggedUser.mock';
import userListMock from '../../mocks/userList.mock';
import userSocialsMock from '../../mocks/userSocials.mock';
import {getUsers} from '../../redux/actions/usersList.creator';
import usersListActions from '../../redux/actions/usersList.actions';

jest.mock('../../redux/actions/usersList.creator', () => ({
  ...jest.requireActual('../../redux/actions/usersList.creator'),
  getUsers: jest.fn(),
}));

const navigation = {};

const route = {
  params: {
    logo: '',
    followingPage: true,
  },
};

describe('Given a UsersList component', () => {
  describe('When it is rendered', () => {
    describe('And userList results is false', () => {
      test('Then should match the snapshot', () => {
        const initialState = {
          loggedUser: {userData: loggedUserMock.user},
          usersList: {users: userListMock, results: false},
          userSocials: userSocialsMock,
        };
        const screen = render(
          <UsersList navigation={navigation} route={route} />,
          initialState,
        );
        expect(screen).toMatchSnapshot();
      });
    });

    describe('And userList results is true', () => {
      describe('And there are no users', () => {
        test('Then should match the snapshot', () => {
          const initialState = {
            loggedUser: {userData: loggedUserMock.user},
            usersList: {users: [], results: true},
            userSocials: userSocialsMock,
          };
          const screen = render(
            <UsersList navigation={navigation} route={route} />,
            initialState,
          );
          expect(screen).toMatchSnapshot();
        });
      });

      describe('And there are users', () => {
        let screen: any;
        beforeEach(() => {
          (getUsers as jest.Mock).mockReturnValue({type: ''});
          const initialState = {
            loggedUser: {userData: loggedUserMock.user},
            usersList: {users: userListMock, results: true},
            userSocials: userSocialsMock,
          };
          screen = render(
            <UsersList navigation={navigation} route={route} />,
            initialState,
          );
        });
        test('Then should match the snapshot', () => {
          expect(screen).toMatchSnapshot();
        });

        describe('And you type on searchInput', () => {
          const typedValue = 'searchThis';
          beforeEach(() => {
            const searchInput = screen.getByTestId('searchInput');
            fireEvent.changeText(searchInput, typedValue);
          });
          test('Then the value typed should be rendered', () => {
            expect(screen.queryByDisplayValue(typedValue)).not.toBe(null);
          });

          describe('And you press searchButton', () => {
            beforeEach(() => {
              const searchButton = screen.getByTestId('searchButton');
              fireEvent.press(searchButton);
            });
            test('Then getUsers should have been called', () => {
              expect(getUsers).toHaveBeenCalled();
            });

            describe('And searchResults are loading', () => {
              test('Then should render an Activity indicator with testID activityIndicator', () => {
                expect(screen.queryByTestId('activityIndicator')).not.toBe(
                  null,
                );
              });
            });

            describe('And searchResults are loaded', () => {
              beforeEach(() => {
                (getUsers as jest.Mock).mockReturnValue({
                  type: usersListActions.LOAD_USERS_LIST,
                  data: userListMock.splice(0, 1),
                });
                const initialState = {
                  loggedUser: {userData: loggedUserMock.user},
                  usersList: {users: userListMock, results: true},
                  userSocials: userSocialsMock,
                };
                screen = render(
                  <UsersList navigation={navigation} route={route} />,
                  initialState,
                );

                const searchInput = screen.getByTestId('searchInput');
                fireEvent.changeText(searchInput, typedValue);
                const searchButton = screen.getByTestId('searchButton');
                fireEvent.press(searchButton);
              });
              test('Then should render a button with testID clearSearchButton', () => {
                expect(screen.queryByTestId('clearSearchButton')).not.toBe(
                  null,
                );
              });

              describe('And you press searchButton with searchInput empty', () => {
                test('Then should match the snapshot', () => {
                  const searchInput = screen.getByTestId('searchInput');
                  fireEvent.changeText(searchInput, '');
                  const searchButton = screen.getByTestId('searchButton');
                  fireEvent.press(searchButton);

                  expect(screen).toMatchSnapshot();
                });
              });

              describe('And you press clearSearchButton', () => {
                test('Then should match the snapshot', () => {
                  const clearSearchButton =
                    screen.getByTestId('clearSearchButton');
                  fireEvent.press(clearSearchButton);

                  expect(screen).toMatchSnapshot();
                });
              });
            });
          });
        });
      });
    });
  });
});
