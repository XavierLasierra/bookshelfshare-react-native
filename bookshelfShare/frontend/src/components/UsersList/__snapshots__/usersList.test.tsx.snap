// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a UsersList component When it is rendered And userList results is false Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": 20,
        "zIndex": 1,
      }
    }
    testID="backButton"
  >
    <IconMock
      height={30}
      style={
        Object {
          "transform": Array [
            Object {
              "rotateZ": "180deg",
            },
          ],
        }
      }
      width={30}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 30,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          maxLength={25}
          onChangeText={[Function]}
          placeholder="Search new users by email/name"
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#ECECEC",
              "borderRadius": 10,
              "borderWidth": 1,
              "fontSize": 16,
              "height": "100%",
              "paddingHorizontal": 20,
              "width": "85%",
            }
          }
          testID="searchInput"
          value=""
        />
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="searchButton"
        >
          <IconMock
            height={35}
            width={35}
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <ActivityIndicator
        color="#000000"
        size="large"
        testID="activityIndicator"
      />
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Given a UsersList component When it is rendered And userList results is true And there are no users Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": 20,
        "zIndex": 1,
      }
    }
    testID="backButton"
  >
    <IconMock
      height={30}
      style={
        Object {
          "transform": Array [
            Object {
              "rotateZ": "180deg",
            },
          ],
        }
      }
      width={30}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 30,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          maxLength={25}
          onChangeText={[Function]}
          placeholder="Search new users by email/name"
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#ECECEC",
              "borderRadius": 10,
              "borderWidth": 1,
              "fontSize": 16,
              "height": "100%",
              "paddingHorizontal": 20,
              "width": "85%",
            }
          }
          testID="searchInput"
          value=""
        />
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="searchButton"
        >
          <IconMock
            height={35}
            width={35}
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Text>
        0 users
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Given a UsersList component When it is rendered And userList results is true And there are users And you type on searchInput And you press searchButton And searchResults are loaded And you press clearSearchButton Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": 20,
        "zIndex": 1,
      }
    }
    testID="backButton"
  >
    <IconMock
      height={30}
      style={
        Object {
          "transform": Array [
            Object {
              "rotateZ": "180deg",
            },
          ],
        }
      }
      width={30}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 30,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          maxLength={25}
          onChangeText={[Function]}
          placeholder="Search new users by email/name"
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#ECECEC",
              "borderRadius": 10,
              "borderWidth": 1,
              "fontSize": 16,
              "height": "100%",
              "paddingHorizontal": 20,
              "width": "85%",
            }
          }
          testID="searchInput"
          value=""
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="searchButton"
        >
          <IconMock
            height={35}
            width={35}
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <RCTScrollView
        ListFooterComponent={[Function]}
        data={
          Array [
            Object {
              "_id": "3",
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsVerticalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={Array []}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#E4FDFF",
                  "borderColor": "#ECECEC",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 10,
                  "opacity": 1,
                  "padding": 10,
                  "width": 325,
                }
              }
              testID="userDetailButton"
            >
              <Image
                source={
                  Object {
                    "uri": undefined,
                  }
                }
                style={
                  Object {
                    "borderRadius": 100,
                    "height": 75,
                    "width": 75,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "width": 200,
                  }
                }
              >
                <View
                  style={Object {}}
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 16,
                        "fontWeight": "bold",
                      }
                    }
                  />
                  <Text
                    style={
                      Object {
                        "fontSize": 16,
                      }
                    }
                  />
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="deleteFollowingButton"
                >
                  <IconMock
                    height={30}
                    width={30}
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
          >
            <View
              style={
                Object {
                  "height": 550,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Given a UsersList component When it is rendered And userList results is true And there are users And you type on searchInput And you press searchButton And searchResults are loaded And you press searchButton with searchInput empty Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": 20,
        "zIndex": 1,
      }
    }
    testID="backButton"
  >
    <IconMock
      height={30}
      style={
        Object {
          "transform": Array [
            Object {
              "rotateZ": "180deg",
            },
          ],
        }
      }
      width={30}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 30,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          maxLength={25}
          onChangeText={[Function]}
          placeholder="Search new users by email/name"
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#ECECEC",
              "borderRadius": 10,
              "borderWidth": 1,
              "fontSize": 16,
              "height": "100%",
              "paddingHorizontal": 20,
              "width": "85%",
            }
          }
          testID="searchInput"
          value=""
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="searchButton"
        >
          <IconMock
            height={35}
            width={35}
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <RCTScrollView
        ListFooterComponent={[Function]}
        data={
          Array [
            Object {
              "_id": "3",
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsVerticalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={Array []}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#E4FDFF",
                  "borderColor": "#ECECEC",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 10,
                  "opacity": 1,
                  "padding": 10,
                  "width": 325,
                }
              }
              testID="userDetailButton"
            >
              <Image
                source={
                  Object {
                    "uri": undefined,
                  }
                }
                style={
                  Object {
                    "borderRadius": 100,
                    "height": 75,
                    "width": 75,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "width": 200,
                  }
                }
              >
                <View
                  style={Object {}}
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 16,
                        "fontWeight": "bold",
                      }
                    }
                  />
                  <Text
                    style={
                      Object {
                        "fontSize": 16,
                      }
                    }
                  />
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="deleteFollowingButton"
                >
                  <IconMock
                    height={30}
                    width={30}
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
          >
            <View
              style={
                Object {
                  "height": 550,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Given a UsersList component When it is rendered And userList results is true And there are users Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 20,
        "opacity": 1,
        "position": "absolute",
        "top": 20,
        "zIndex": 1,
      }
    }
    testID="backButton"
  >
    <IconMock
      height={30}
      style={
        Object {
          "transform": Array [
            Object {
              "rotateZ": "180deg",
            },
          ],
        }
      }
      width={30}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 30,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          maxLength={25}
          onChangeText={[Function]}
          placeholder="Search new users by email/name"
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#ECECEC",
              "borderRadius": 10,
              "borderWidth": 1,
              "fontSize": 16,
              "height": "100%",
              "paddingHorizontal": 20,
              "width": "85%",
            }
          }
          testID="searchInput"
          value=""
        />
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="searchButton"
        >
          <IconMock
            height={35}
            width={35}
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <RCTScrollView
        ListFooterComponent={[Function]}
        data={
          Array [
            Object {
              "_id": "1",
              "email": "xlp1998@gmail.com",
              "photo": "photo",
              "username": "Xavier Lasierra",
            },
            Object {
              "_id": "2",
              "email": "mail@gmail.com",
              "photo": "photo",
              "username": "Juan",
            },
            Object {
              "_id": "3",
              "email": "mail@gmail.com",
              "photo": "photo",
              "username": "Jose",
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsVerticalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={Array []}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          />
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#ECECEC",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 10,
                  "opacity": 1,
                  "padding": 10,
                  "width": 325,
                }
              }
              testID="userDetailButton"
            >
              <Image
                source={
                  Object {
                    "uri": "photo",
                  }
                }
                style={
                  Object {
                    "borderRadius": 100,
                    "height": 75,
                    "width": 75,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "width": 200,
                  }
                }
              >
                <View
                  style={Object {}}
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 16,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Juan
                  </Text>
                  <Text
                    style={
                      Object {
                        "fontSize": 16,
                      }
                    }
                  >
                    mail@gmail.com
                  </Text>
                </View>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  nativeID="animatedComponent"
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="addFollowingButton"
                >
                  <IconMock
                    height={30}
                    width={30}
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#E4FDFF",
                  "borderColor": "#ECECEC",
                  "borderRadius": 20,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 10,
                  "opacity": 1,
                  "padding": 10,
                  "width": 325,
                }
              }
              testID="userDetailButton"
            >
              <Image
                source={
                  Object {
                    "uri": "photo",
                  }
                }
                style={
                  Object {
                    "borderRadius": 100,
                    "height": 75,
                    "width": 75,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "width": 200,
                  }
                }
              >
                <View
                  style={Object {}}
                >
                  <Text
                    style={
                      Object {
                        "fontSize": 16,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Jose
                  </Text>
                  <Text
                    style={
                      Object {
                        "fontSize": 16,
                      }
                    }
                  >
                    mail@gmail.com
                  </Text>
                </View>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  nativeID="animatedComponent"
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="deleteFollowingButton"
                >
                  <IconMock
                    height={30}
                    width={30}
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
          >
            <View
              style={
                Object {
                  "height": 550,
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTSafeAreaView>
`;
